### ISSUE TITLE
SwiftUI Previews are not working

### ISSUE THREAD

**Issue by dancarvajc** [akaffenberger/firebase-ios-sdk-xcframeworks#46]
Hello, thanks for creating this package, it really helps with build times. Just one problem, I can't get the Swiftui Previews to work. The official firebase package works without problems. Do you know if it is something specific with this mirror?

**Comment by alexfringes** [comment]
FWIW, I have no issue with Previews with this project included. But there is apparently an ongoing issue with Previews and binary SPM: [Discussion on Dev Forums](https://developer.apple.com/forums/thread/707569)

**Comment by sisoje** [comment]
i think it works in 14.1 beta2

**Comment by akaffenberger** [comment]
Which libraries are you using? I can try to repro by adding some more examples to the iOS Example project.

From my testing so far, certain libraries work with previews while others don't. ie Google Sign In works, In app messaging works, Firestore crashes it. However, the Firestore crash I can repro with the official package too, so that might be a different issue.

**Comment by sisoje** [comment]
it works in xcode 14.1

**Comment by dushantSW** [comment]
We are seeing a "Preview crashed" issue. Using FirebaseAnalytics, Messaging, Functions & Performance

**Comment by akaffenberger** [comment]
On XCode 14.2 I'm still seeing a previews crash for Firestore (in both this repo and the official repo), but Messaging and Google Sign In still works fine. When I find some time I'll give Analytics, Functions, and Performance a try.

**Comment by js** [comment]
I also get the "Preview crashed" message. Looking at the diagnostics in ` ~/Library/Logs/DiagnosticReports`:

```
Last Exception Backtrace:
0   CoreFoundation                	       0x18040e7bc __exceptionPreprocess + 160
1   libobjc.A.dylib               	       0x180051144 objc_exception_throw + 56
2   CoreFoundation                	       0x18041d47c +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
3   CoreFoundation                	       0x1804126c8 ___forwarding___ + 1308
4   CoreFoundation                	       0x180414b4c _CF_forwarding_prep_0 + 92
5   Mentra                        	       0x103b7ba14 -[GDTCCTUploadOperation updateNextUploadTimeWithResponse:forTarget:] + 64
6   Mentra                        	       0x103b7b058 __54-[GDTCCTUploadOperation uploadBatch:toTarget:storage:]_block_invoke + 72
7   Mentra                        	       0x103a0e0f0 __56-[FBLPromise chainOnQueue:chainedFulfill:chainedReject:]_block_invoke.48 + 52
8   libdispatch.dylib             	       0x180133fa4 _dispatch_call_block_and_release + 24
9   libdispatch.dylib             	       0x180135768 _dispatch_client_callout + 16
10  libdispatch.dylib             	       0x18013d9d8 _dispatch_lane_serial_drain + 1260
11  libdispatch.dylib             	       0x18013e43c _dispatch_lane_invoke + 392
12  libdispatch.dylib             	       0x18014a624 _dispatch_workloop_worker_thread + 772
13  libsystem_pthread.dylib       	       0x1b0598b40 _pthread_wqthread + 284
14  libsystem_pthread.dylib       	       0x1b0597904 start_wqthread + 8
```

Which seems to be the same crash [mentioned here](https://github.com/firebase/firebase-ios-sdk/issues/8005). However if I integrate the Firebase libraries using their official SPM package the previews doesn't crash, but the time it takes to compile FirestoreStore is close to unbearable though, which is why we switched to this package to begin with :)

**Comment by MartinPrusa** [comment]
i think this could be closed, the issue disappeared in Firebase 11.0.0

**Comment by bill-foreflight** [comment]
The issue improved with the 11 series but continues for us. It no longer fails 100% of the time but still fails about 50% of the time. The SwiftUI preview will either never load or just show a blank view. We can't use the library unfortunately anymore due to this. I'd suggest this issue be reopened.

This can be demonstrated in a minimal sample project with the following.

Hardware: Mac Pro with M1

Package.swift:

```
// swift-tools-version: 5.9
// The swift-tools-version declares the minimum version of Swift required to build this package.

import Foundation
import PackageDescription
import CompilerPluginSupport

let package = Package(
    name: "FlagKit",
    platforms: [.iOS(.v17), .macOS(.v13), .watchOS(.v10)],
    products: [
        .library( 
            name: "FlagKit", // In this target, use the ConfigUpdateListenerRegistration symbol.
            targets: ["FlagKit"]
        ),
        .library(
            name: "FlagKitUI", // In this target, add a "Hello World" SwiftUI file with previews.
            targets: ["FlagKitUI"]
        )
    ],
    dependencies: [
        .package(url: "https://github.com/akaffenberger/firebase-ios-sdk-xcframeworks.git", exact: "11.9.0"),
    ],
    targets: [
        .target(
            name: "FlagKit",
            dependencies: [
                .product(name: "FirebaseRemoteConfig", package: "firebase-ios-sdk-xcframeworks"),
            ]
        ),
        .target(
            name: "FlagKitUI",
            dependencies: [
                "FlagKit"
            ]
        ),
    ]
)
```
